Codecademy/gamut
# ci: migrate codebase to use NX (using lerna integration)

### Overview


Switch to NX for task caching and package builds, further integration in future updates


Next steps

- [ ] convert repo to use nx test utilities & cache in CI
- [ ] convert repo to use nx lint utilities & cache in CI

### PR Checklist


#### Testing Instructions

To test the dev experience, pull down this branch and run `yarn build`, then run it again. The second time everything should be cached and the build should be basically instant.

Now change a package that is shared by a few other packages, like `@codecademy/gamut-illustrations`, and run `yarn build` again. You should see `gamut`, `gamut-labs`, and `monaco` are rebuilt, and all other builds are cached.

Everything should be the same for the builds, i didn't modify how the packages were being built, just what was building them. you can confirm by comparing the contents of the latest version of the packages with the alpha versions from this PR.

https://unpkg.com/browse/@codecademy/gamut-icons@7.4.3-alpha.7b7eab.0/dist/icons/regular/AccountingCoinsIcon.js
https://unpkg.com/browse/@codecademy/gamut-icons@7.4.2/dist/icons/regular/AccountingCoinsIcon.js



#### PR Links and Envs


| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/31722)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
